app/decorators/course_decorator.rb

```diff
@@ -214,6 +214,13 @@ def selectable_subjects
     meta["edit_options"]["subjects"].map { |subject| [subject["attributes"]["subject_name"], subject["id"]] }
   end
 
+  def selected_subject_ids
+    selectable_subject_ids = meta["edit_options"]["subjects"].map { |subject| subject["id"] }
+    selected_subject_ids = subjects.map(&:id)
+
+    selectable_subject_ids & selected_subject_ids
+  end
+
```

Comment:
https://github.com/DFE-Digital/publish-teacher-training/blob/master/spec/decorators/course_decorator_spec.rb

Could do with adding tests for this :eyes: